diff --git a/dist/copilotkit-JdHy0xdu.mjs b/dist/copilotkit-JdHy0xdu.mjs
index 0d27e159b007d8fbdeb7d316f13f3c30bb6e38be..4c4b8a48d9d7e4fc1285fcade5b8699f4d7a2c2a 100644
--- a/dist/copilotkit-JdHy0xdu.mjs
+++ b/dist/copilotkit-JdHy0xdu.mjs
@@ -12069,16 +12069,7 @@ function entryPointsToFunctionCallHandler(actions) {
 		const action = actionsByFunctionName[name];
 		let result = void 0;
 		if (action) {
-			await new Promise((resolve, reject) => {
-				flushSync(async () => {
-					try {
-						result = await action.handler?.(args);
-						resolve();
-					} catch (error) {
-						reject(error);
-					}
-				});
-			});
+			result = await action.handler?.(args);
 			await new Promise((resolve) => setTimeout(resolve, 20));
 		}
 		return result;
